test/e2e/tests/releases/utils/methods.ts

```diff
@@ -0,0 +1,211 @@
+  dataset,
+  releaseId,
+}: {
+  sanityClient: SanityClient
+  dataset: string | undefined
+  releaseId: string
+}) => {
+  // delete release
+  sanityClient.withConfig(CLIENT_OPTIONS).request({
+    uri: `/data/actions/${dataset}`,
+    method: 'POST',
+    body: {
+      actions: [
+        {
+          actionType: 'sanity.action.release.delete',
+          releaseId: releaseId,
+        },
+      ],
+    },
+  })
+}
+
+export const archiveAndDeleteRelease = async ({
+  sanityClient,
+  dataset,
+  releaseId,
+}: {
+  sanityClient: SanityClient
+  dataset: string | undefined
+  releaseId: string
+}) => {
+  await archiveRelease({sanityClient, dataset, releaseId})
+  await waitForReleaseToBeArchived({sanityClient, releaseId})
+  await deleteRelease({sanityClient, dataset, releaseId})
+}
+
+export const discardVersion = ({
+  sanityClient,
+  dataset,
+  versionId,
+}: {
+  sanityClient: SanityClient
+  dataset: string | undefined
+  versionId: string
+}) => {
+  // discard release
+  sanityClient.withConfig(CLIENT_OPTIONS).request({
+    uri: `/data/actions/${dataset}`,
+    method: 'POST',
+    body: {
+      actions: [
+        {
+          actionType: 'sanity.action.document.discard',
+          draftId: versionId,
+        },
+      ],
+    },
+  })
+}
+
+/**
+ * @returns a random string of letters and numbers (9 characters long)
+ * @internal
+ */
+export function getRandomReleaseId() {
+  return Math.random().toString(36).slice(2)
+}
+
+export const createDocument = (
+  sanityClient: SanityClient,
+  documentMetada: Partial<SanityDocument> & {_id: string; _type: string},
+) => {
+  return sanityClient.withConfig(CLIENT_OPTIONS).create({
+    ...documentMetada,
+  })
+}
+
+export const deleteAllReleases = async ({
+  sanityClient,
+  dataset,
+}: {
+  sanityClient: SanityClient
+  dataset: string | undefined
+}) => {
+  const query = `*[_type == "system.release"]`
+  const releases = await sanityClient.fetch(query)
+
+  if (!Array.isArray(releases) || releases.length === 0) {
+    console.warn('No releases found to delete.')
+    return
+  }
+
+  try {
+    await Promise.all(
+      releases.map(async (release: {_id: string}) => {
+        const releaseId = release._id.replace('_.releases.', '')
+        return archiveAndDeleteRelease({sanityClient, dataset, releaseId})
+      }),
+    )
+  } catch (error) {
+    console.error('Error deleting releases:', error)
+    throw error
+  }
+}
+
+/**
+ * Polls the Sanity release status until it is 'archived'
+ */
+async function waitForReleaseToBeArchived({
+  sanityClient,
+  releaseId,
+  interval = 5000,
+  timeout = 60000,
+}: {
+  sanityClient: SanityClient
+  releaseId: string
+  interval?: number
+  timeout?: number
+}) {
+  const startTime = Date.now()
+
+  return new Promise<void>((resolve, reject) => {
+    // eslint-disable-next-line consistent-return
+    const checkStatus = async () => {
+      const query = `*[_type == "system.release" && _id == "_.releases.${releaseId}"][0]`
```

Comment:
```suggestion
      const query = `*[_type == "system.release" && _id == "_.releases.${releaseId}"][0] {state}`
```
Adds a projection on `state` since that's all we care about here :)